features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -87,21 +87,24 @@ class Base64Serializer private constructor(
     }
 
     @WorkerThread
-    private fun convertBmpToBase64(bitmap: Bitmap): String {
-        val byteArrayOutputStream = webPImageCompression.compressBitmapToStream(bitmap)
+    private fun convertBmpToBase64(initialBitmap: Bitmap): String {
```

Comment:
in this flow we create 2 bitmaps: first bitmap to draw content of drawable, second bitmap to downscale. I'm wondering if we can do this in 1 pass: can we create a canvas which will apply affine transformation (scale)?

In theory downscaling is just applying the matrix, so maybe we can create a downscaled bitmap straight away (we know size of the drawable, so we should be able to do it), and then provide a canvas which will apply the necessary transformation? This would greatly reduce memory usage.